### ISSUE TITLE
[FEATURE] Upgrade ExoPlayer to 2.17.0

### ISSUE THREAD

**Issue by naamapps** [jhomlala/betterplayer#961]
**Is your feature request related to a problem? Please describe.**
Yes, currently if the package depends both on video_player and betterplayer, we can't build our app because of version mismatch.

**Describe the solution you'd like**
Upgrade exoplayer to the same version video_player uses (2.17.0).

**Describe alternatives you've considered**
Can't think of an alternative.

**Additional context**
This is the error
```
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':app:checkDebugDuplicateClasses'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.CheckDuplicatesRunnable
   > Duplicate class com.google.android.exoplayer2.ui.DownloadNotificationHelper found in modules jetified-exoplayer-core-2.17.0-runtime (com.google.android.exoplayer:exoplayer-core:2.17.0) and jetified-exoplayer-ui-2.15.1-runtime (com.google.android.exoplayer:exoplayer-ui:2.15.1)
```

**Comment by Colton127** [comment]
We need a maintainer to update this plugin as dependencies update (i.e., ExoPlayer). Better_player is in a good state as it is, it just needs some basic maintenance to stay relevant. I opened a pull request for a small bug fix two months ago, and it seems like the package has been abandoned for now, so I don't know if a pull request would be accepted.

**Comment by shiyiya** [comment]
try https://github.com/shiyiya/betterplayer .
- [upgrade android dependencies.
](https://github.com/shiyiya/betterplayer/commit/d5879a5963d9377a97bb7fb064cd35e87311876c)

**Comment by naamapps** [comment]
@shiyiya Thanks for your effort.
However, it is not recommended to use git dependencies in flutter.

**Comment by naamapps** [comment]
@jhomlala Please advise on what should we do here.

**Comment by ayezarniaung** [comment]
> 

I've facing same issue . How to solve it . I also upgrade video player 2.3.0 but can't resolve this problem.

**Comment by hplongqt** [comment]
> 

I tried it, but I can't play video m3u8, the black screen will display. this problem happens android.

**Comment by hplongqt** [comment]
> try https://github.com/shiyiya/betterplayer .
> 
> * [upgrade android dependencies.
>   ](https://github.com/shiyiya/betterplayer/commit/d5879a5963d9377a97bb7fb064cd35e87311876c)
I tried it, but I can't play video m3u8, the black screen will display. this problem happens android.

**Comment by shiyiya** [comment]
@hplongqt could u provide m3u8 url ？

**Comment by hplongqt** [comment]
> @hplongqt could u provide m3u8 url ？
This is url:
https://cdn.theoplayer.com/video/elephants-dream/playlist.m3u8

**Comment by doug-mrowl** [comment]
A temporary fix for this is to add an explicit reference in your app's "pubspec.yaml" file, in it's dependancies section, to the slightly older version of "video_player_android":

  # remove once better_player updates to exoplayer 2.17!
  video_player_android: 2.3.1

This is a deep Flutter dependency of better_player that got updated a few days ago (to version 2.3.2)... I spent hours yesterday trying to figure out this Android build issue, and this got my builds working again.

**Comment by daveshirman** [comment]
@doug-mrowl Thanks for this. Perhaps you could help me understand how/why this is?

So I just started a new app from the cmd line, then copy/pasted in the dependencies from `pubspec.yaml` and encountered this error.  The previous app is still building fine, without the extra `video_player_android` dependency.

Since I haven't upgraded the flutter engine on my machine (still v2.10.2), is this then due to how the Android native project is set up when starting a new flutter app from the cmd line or what? 

I don't see `video_player_android` dependency anywhere on the `better_player` pub.dev page.

Thanks for any input.

**Comment by jhomlala** [comment]
Fixed in latest release (0.0.82).
